internal/server/server.go

```diff
@@ -65,19 +65,32 @@ func New(logger log.Logger, reg *prometheus.Registry, opts ...Option) Server {
 
 	registerProber(r, p)
 
-	uiPath := "/ui/metrics/v1"
-
-	r.Get("/", func(w http.ResponseWriter, r *http.Request) {
-		http.Redirect(w, r, path.Join(uiPath, "graph"), http.StatusMovedPermanently)
-	})
-
 	r.Get("/metrics", func(w http.ResponseWriter, r *http.Request) {
 		promhttp.InstrumentMetricHandler(reg, promhttp.HandlerFor(reg, promhttp.HandlerOpts{})).ServeHTTP(w, r)
 	})
 
 	if options.metricsUIEndpoint != nil {
+		uiPath := "/ui/metrics/v1"
+
 		r.Get(path.Join(uiPath, "*"),
 			ins.newHandler("ui", proxy.New(logger, uiPath, options.metricsUIEndpoint, options.proxyOptions...)))
+
+		r.Get("/", func(w http.ResponseWriter, r *http.Request) {
+			http.Redirect(w, r, path.Join(uiPath, "graph"), http.StatusMovedPermanently)
+		})
+
+		// NOTICE: Following redirects added to be compatible with existing Read UI.
+		// Paths are explicitly specified to prevent unnecessary request to read handler.
+		for _, p := range []string{
+			"graph",
+			"stores",
+			"status",
+		} {
+			p := p
```

Comment:
why do we need this extra declaration here? It looks to me like all uses of `p` are synchronous with the loop

Reply:
If you see line 91, we are using it in a closure because of that usage we need to re-declare it in the loop scope.

Reply:
ack yes I see it now